src/diffusers/pipelines/consistency_models/pipeline_consistency_models.py

```diff
@@ -0,0 +1,312 @@
+                "`pil`, `np`, `pt`, `latent`"
+            )
+            deprecate("Unsupported output_type", "1.0.0", deprecation_message, standard_warn=False)
+            output_type = "np"
+
+        if output_type == "latent":
+            # Return latents without modification
+            return latents
+
+        # Equivalent to diffusers.VaeImageProcessor.denormalize
+        latents = (latents / 2 + 0.5).clamp(0, 1)
+        if output_type == "pt":
+            return latents
+
+        # Equivalent to diffusers.VaeImageProcessor.pt_to_numpy
+        latents = latents.cpu().permute(0, 2, 3, 1).numpy()
+        if output_type == "np":
+            return latents
+
+        # Output_type must be 'pil'
+        latents = self.numpy_to_pil(latents)
+        return latents
+
+    def prepare_class_labels(self, batch_size, device, class_labels=None):
+        if self.unet.config.num_class_embeds is not None:
+            if isinstance(class_labels, list):
+                class_labels = torch.tensor(class_labels, dtype=torch.int)
+            elif isinstance(class_labels, int):
+                assert batch_size == 1, "Batch size must be 1 if classes is an int"
+                class_labels = torch.tensor([class_labels], dtype=torch.int)
+            elif class_labels is None:
+                # Randomly generate batch_size class labels
+                # TODO: should use generator here? int analogue of randn_tensor is not exposed in ...utils
+                class_labels = torch.randint(0, self.unet.config.num_class_embeds, size=(batch_size,))
+            class_labels = class_labels.to(device)
+        else:
+            class_labels = None
+        return class_labels
+
+    def check_inputs(self, num_inference_steps, timesteps, latents, batch_size, img_size, callback_steps):
+        if num_inference_steps is None and timesteps is None:
+            raise ValueError("Exactly one of `num_inference_steps` or `timesteps` must be supplied.")
+
+        if num_inference_steps is not None and timesteps is not None:
+            logger.warning(
+                f"Both `num_inference_steps`: {num_inference_steps} and `timesteps`: {timesteps} are supplied;"
+                " `timesteps` will be used over `num_inference_steps`."
+            )
+
+        if latents is not None:
+            expected_shape = (batch_size, 3, img_size, img_size)
+            if latents.shape != expected_shape:
+                raise ValueError(f"The shape of latents is {latents.shape} but is expected to be {expected_shape}.")
+
+        if (callback_steps is None) or (
+            callback_steps is not None and (not isinstance(callback_steps, int) or callback_steps <= 0)
+        ):
+            raise ValueError(
+                f"`callback_steps` has to be a positive integer but is {callback_steps} of type"
+                f" {type(callback_steps)}."
+            )
+
+    @torch.no_grad()
```

Comment:
Maybe also makes sense to add a decorator to have some example usage here. Example:

https://github.com/huggingface/diffusers/blob/2837d490799e33aee66be50697d06d2adb61d0aa/src/diffusers/pipelines/deepfloyd_if/pipeline_if.py#L626